components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/IdentifierHandler.java

```diff
@@ -593,4 +576,97 @@ private void persistUsername(AuthenticationContext context, String username) {
         context.getPreviousAuthenticatedIdPs().clear();
         context.addAuthenticatorParams(contextParams);
     }
+
+    /**
+     * Validate the username against the user store.
+     * If not found in the PRIMARY userstore and the username is not domain qualified,
+     * then search in secondary userstores.
+     *
+     * @param tenantDomain          Tenant domain.
+     * @param username              Username of the user.
+     * @param tenantAwareUsername   Tenant aware username.
+     * @param identifierFromRequest Identifier provided in the request.
+     * @param userId                User id if present.
+     * @return User id and user store domain (If found from secondary user stores).
+     * @throws AuthenticationFailedException If user not found or an error happens.
+     */
+    private String[] validateUsername(String tenantDomain, String username, String tenantAwareUsername,
+                                  String identifierFromRequest, String userId)
+            throws AuthenticationFailedException {
+
+        AbstractUserStoreManager userStoreManager;
+        String userStoreDomain = null;
+        // Check for the username exists.
+        try {
+            int tenantId = IdentifierAuthenticatorServiceComponent
+                    .getRealmService().getTenantManager().getTenantId(tenantDomain);
+            UserRealm userRealm = IdentifierAuthenticatorServiceComponent.getRealmService()
+                    .getTenantUserRealm(tenantId);
+
+            if (userRealm != null) {
```

Comment:
can invert the if condition and get rid of else check for better readability

Reply:
Fixed